### ISSUE TITLE
Redirect pattern not working with a bucket item

### ISSUE THREAD

**Issue by gonzalocalandria-xcentium** [thecadams/301RedirectModule#26]
Hello, i'm having a problem with the redirect pattern item.

This is my setup:
Requested Expression: `^/assets/files/news/press_releases/.*\.pdf`
Response Status Code: 301
Source Item: `/sitecore/content/Home/News and Community/Press-Releases` 

The goal: all pdf's under `/assets/files/news/press_releases/.../*.pdf` are deleted so i'm redirecting all those to the route `/News and Community/Press-Releases` 
The problem: the pdf's url's are giving 404 instead the redirection. i.e.: `http://local.mysite.com/assets/files/news/press_releases/2016_PR/Custom-Release.pdf`

Some information: 

- The `/sitecore/content/Home/News and Community/Press-Releases`  is a bucket  
- if i point the Source Item to `News and Community` the redirect works
- if if enter on the address bar `http://local.mysite.com/news-and-community/Press-Releases` it works perfectly.
- The .pdf extension is included on `Sitecore.config` allowed extensions.

So.. what's going on? do i have something wrong?

**Comment by thecadams** [comment]
Hi! Unfortunately you may be the first person to try and use the module with bucket items.

My suspicion is the 301 module doesn't search the bucket for the item you're hoping to find - without calling the bucket API, the bucket looks empty. If that's the case, it's no problem if you get it working and would like to put together a pull request - after all, you're unlikely to be the last to try to use bucket items with redirects.

**Comment by gonzalocalandria-xcentium** [comment]
But if I set the same redirect but with a Redirect Url instead a Redirect Pattern, the redirect works... maybe this change what you think that may be not working?

**Comment by thecadams** [comment]
I wonder if the Item Resolver is getting involved in that case and resolving the item for you, because the regex matching is string based?

**Comment by gonzalocalandria-xcentium** [comment]
Actually I don't know, I'm quite new on sitecore. One thing that I can say is that there is a 404 pipeline handler. But I don't know if that can be affecting the module.

**Comment by thecadams** [comment]
Ok! Well one way to test this hypothesis is to delete the 494 handler from your config or comment it out. Try that and let me know if it helps!

**Comment by gonzalocalandria-xcentium** [comment]
I've deactivated 404 module and it throws the default document not found sitecore page, so is not the 404 module what is affecting the Redirect Pattern.
Did you find something on the module that could be causing this?

**Comment by thecadams** [comment]
Hi, little update here. My Sitecore license has expired but as soon as I hear back from Sitecore sales I'll be able to test for you. Did you end up solving the issue though?

**Comment by gonzalocalandria-xcentium** [comment]
No we didn't fix the problem

**Comment by thecadams** [comment]
cc @Chris-Castle I'm still waiting for a Sitecore license so I can test this.

**Comment by angusnml** [comment]
I don't think this issue has to do with buckets. I'm having a similar issue with Regex redirects when the source item it's redirecting to has a dash in the path. It works 100% when the source item path doesn't contain a dash.

**Comment by christianwccc** [comment]
Hi guys, 

Holy dead thread, I know.
angusnml has it right.

To elaborate, this setting from web.config in <encodeNameReplacements>-section is causing the issue.
`<replace mode="on" find=" " replaceWith="-">`

This causes the DecodeName function to replace the dash with a whitespace (so just to clarify, the dash is not "stripped" as suggested by angusnml, but rather replaced with a whitespace, in accordance with the default Sitecore configuration).

This took me a while to realise. 
In our case, we handle whitespaces another way, so we were able to simply remove that line from the configuration, and then it worked.
